lark/parsers/earley.py

```diff
@@ -30,7 +30,8 @@ class Parser:
     debug: bool
 
     def __init__(self, lexer_conf: 'LexerConf', parser_conf: 'ParserConf', term_matcher: Callable,
-                 resolve_ambiguity: bool=True, debug: bool=False, tree_class: type=Tree, ordered_sets: bool=True):
+                 resolve_ambiguity: bool=True, debug: bool=False,
+                 tree_class: Optional[type]=Tree, ordered_sets: bool=True):
```

Comment:
Did you mean to make it `Optional[Callable[[str, Sequence], Any]]`?
Same question for `xearley.py`